x-pack/legacy/plugins/transform/public/shim.ts

```diff
@@ -0,0 +1,149 @@
+
+export interface AppCore {
+  chrome: typeof chrome;
+  i18n: {
+    [i18nPackage: string]: any;
+    Context: typeof I18nContext;
+    FormattedMessage: typeof FormattedMessage;
+    FormattedDate: typeof FormattedDate;
+    FormattedTime: typeof FormattedTime;
+  };
+  notification: {
+    fatalError: typeof fatalError;
+    toastNotifications: typeof toastNotifications;
+  };
+  savedSearches: {
+    getClient(): any;
+    setClient(client: any): void;
+  };
+}
+
+export interface AppPlugins {
+  management: {
+    sections: typeof management;
+  };
+}
+
+export interface Core extends AppCore {
+  chrome: typeof chrome;
+  http: {
+    getClient(): any;
+    setClient(client: any): void;
+  };
+  routing: {
+    registerAngularRoute(path: string, config: object): void;
+    registerRouter(router: HashRouter): void;
+    getRouter(): HashRouter | undefined;
+  };
+  documentation: {
+    esDocBasePath: string;
+    esPluginDocBasePath: string;
+    esStackOverviewDocBasePath: string;
+  };
+  docTitle: {
+    change: typeof docTitle.change;
+  };
+}
+
+export interface Plugins extends AppPlugins {
+  management: {
+    sections: typeof management;
+    constants: {
+      BREADCRUMB: typeof MANAGEMENT_BREADCRUMB;
+    };
+  };
+  uiMetric: {
+    createUiStatsReporter: typeof createUiStatsReporter;
+  };
+}
+
+export function createPublicShim(): { core: Core; plugins: Plugins } {
+  // This is an Angular service, which is why we use this provider pattern
+  // to access it within our React app.
+  let httpClient: ng.IHttpService;
+  // This is an Angular service, which is why we use this provider pattern
+  // to access it within our React app.
+  let savedSearches: SavedSearchLoader;
+
+  let reactRouter: HashRouter | undefined;
+
+  return {
+    core: {
+      i18n: {
+        ...i18n,
+        Context: I18nContext,
+        FormattedMessage,
+        FormattedDate,
+        FormattedTime,
+      },
+      routing: {
+        registerAngularRoute: (path: string, config: object): void => {
+          routes.when(path, config);
+        },
+        registerRouter: (router: HashRouter): void => {
+          reactRouter = router;
+        },
+        getRouter: (): HashRouter | undefined => {
+          return reactRouter;
+        },
+      },
+      http: {
+        setClient: (client: any): void => {
+          httpClient = client;
+        },
+        getClient: (): any => httpClient,
+      },
+      chrome,
+      notification: {
+        fatalError,
+        toastNotifications,
+      },
+      documentation: {
+        esDocBasePath: `${ELASTIC_WEBSITE_URL}guide/en/elasticsearch/reference/${DOC_LINK_VERSION}/`,
+        esPluginDocBasePath: `${ELASTIC_WEBSITE_URL}guide/en/elasticsearch/plugins/${DOC_LINK_VERSION}/`,
+        esStackOverviewDocBasePath: `${ELASTIC_WEBSITE_URL}guide/en/elastic-stack-overview/${DOC_LINK_VERSION}/`,
+      },
+      docTitle: {
+        change: docTitle.change,
+      },
+      savedSearches: {
```

Comment:
savedSearches will come via plugins rather than via core.

Reply:
Moved to `plugins` in e1320e6.